frontend/src/store/FamLoginUserState.ts

```diff
@@ -215,66 +228,93 @@ const getCachedAppRolesForDelegatedAdmin = (
-                application: famGrant?.application.description,
-                env: famGrant?.application.env,
-                role: 'Admin',
-            });
-        }
+    return [
+        {
+            application: famGrant.application.description,
+            env: famGrant.application.env,
+            role: 'Admin',
+        },
+    ];
+};
 
-        if (item.auth_key === AdminRoleAuthGroup.AppAdmin) {
-            item.grants.forEach((grant) => {
-                myPermissions.push({
-                    role: 'Admin',
+const getMyAppAdminPermission = (access: FamAuthGrantDto): IMyPermission[] => {
+    return access.grants.map((grant: FamGrantDetailDto) => ({
+        application: grant.application.description,
+        env: grant.application.env,
+        role: 'Admin',
+    }));
+};
+
+const getMyDelegatedAdminPermission = (
+    access: FamAuthGrantDto
+): IMyPermission[] => {
+    const permissions: IMyPermission[] = [];
+    access.grants.forEach((grant: FamGrantDetailDto) => {
+        grant.roles?.forEach((role: FamRoleDto) => {
+            const roleDescription = 'Delegated Admin, ' + role.name;
+            if (!role.forest_clients) {
+                permissions.push({
                     application: grant.application.description,
                     env: grant.application.env,
+                    role: roleDescription,
                 });
-            });
-        }
-
-        if (item.auth_key === AdminRoleAuthGroup.DelegatedAdmin) {
-            item.grants.forEach((grant) => {
-                grant.roles?.forEach((role) => {
-                    if(!role.forest_clients) {
-                        myPermissions.push({
-                            application: grant.application.description,
-                            env: grant.application.env,
-                            clientId: null,
-                            role: 'Delegated Admin, ' + role.name,
-                        });
-                    } else {
-                        role.forest_clients?.forEach((clientId) => {
-                            myPermissions.push({
-                                application: grant.application.description,
-                                env: grant.application.env,
-                                clientId: clientId,
-                                role: 'Delegated Admin, ' + role.name,
-                            });
-                        });
-                    }
+            } else {
+                role.forest_clients.forEach((clientId: any) => {
+                    permissions.push({
+                        application: grant.application.description,
+                        env: grant.application.env,
+                        clientId: clientId,
+                        role: roleDescription,
+                    });
                 });
-            });
+            }
+        });
+    });
+
+    return permissions;
+};
+
+const getMyAdminPermission = (): IMyPermission[] => {
```

Comment:
Can we have this method below the last method of the "// --- getters" area (which is for the getters of the state and they are public/visible)?
The intention was to organize them and not being placed all over the place, to be easier to find.